projects/pswg/src/main/java/com/parzivail/pswg/item/ThermalDetonatorItem.java

```diff
@@ -0,0 +1,177 @@
+package com.parzivail.pswg.item;
+
+import com.parzivail.pswg.container.SwgEntities;
+import com.parzivail.pswg.container.SwgItems;
+import com.parzivail.pswg.entity.ThermalDetonatorEntity;
+import com.parzivail.util.item.ILeftClickConsumer;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.network.packet.s2c.play.EntitySpawnS2CPacket;
+import net.minecraft.sound.SoundCategory;
+import net.minecraft.sound.SoundEvents;
+import net.minecraft.stat.Stats;
+import net.minecraft.text.Text;
+import net.minecraft.util.Hand;
+import net.minecraft.util.TypedActionResult;
+import net.minecraft.util.UseAction;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.Nullable;
+
+public class ThermalDetonatorItem extends Item implements ILeftClickConsumer
+{
+	public ThermalDetonatorItem(Settings settings)
+	{
+		super(settings);
+	}
+	public ThermalDetonatorEntity createThermalDetonator(World world, ItemStack stack, int life, boolean primed, PlayerEntity playerEntity) {
+		ThermalDetonatorEntity thermalDetonatorEntity = new ThermalDetonatorEntity(SwgEntities.Misc.ThermalDetonator, world);
+		thermalDetonatorEntity.setLife(life);
+		thermalDetonatorEntity.setPrimed(primed);
+		thermalDetonatorEntity.onSpawnPacket(new EntitySpawnS2CPacket(thermalDetonatorEntity.getId(), thermalDetonatorEntity.getUuid(), playerEntity.getX(), playerEntity.getY()+1, playerEntity.getZ(), -playerEntity.getPitch(), -playerEntity.getYaw(), thermalDetonatorEntity.getType(),0, new Vec3d(0f, 0f ,0f), playerEntity.getHeadYaw()));
+		return thermalDetonatorEntity;
+	}
+	int ticksToExplosion=1000000;
```

Comment:
These should be part of an NBT tag. Only one instance of the `ThermalDetonatorItem` class is ever created, so these are effectively shared globally between all itemstacks. See `BlasterTag` or `LightsaberTag` to get an idea on how those items store strongly-typed stack data